presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1499,4 +1514,14 @@ private static void quoteStringLiteral(StringBuilder out, String value)
         }
         out.append('\'');
     }
+
+    private boolean stringIsLikePattern(String value) throws SQLException
+    {
+        if (value == null) {
+            return false;
+        }
+        Pattern p = Pattern.compile("[%_]");
```

Comment:
If you remove all occurrences of escape together with next char (thus all even consecutive occurrences or escape too) with `value.replaceAll(Pattern.quote(getSearchStringEscape()) + ".", "")`, then testing with `Pattern.compile("[%_]")` (or equivalently `value.contains("%") || value.contains("_")`) tells whether it was LIKE pattern or not.